From 3c7fe585eb4f238288b28192b80af128e91b8fe7 Mon Sep 17 00:00:00 2001
From: Chaitanya Kulkarni <chaitanya.kulkarni@wdc.com>
Date: Fri, 7 Aug 2020 20:52:15 -0700
Subject: [PATCH 09/13] nvmet: prep patch for pt ctrl put wrapper

This is a preparation patch to cnetralize the passthru put controller
action. Next patch actully fixes the bug in passthru code which needs
this helper to avoid the duplication of the code.

Signed-off-by: Chaitanya Kulkarni <chaitanya.kulkarni@wdc.com>
---
 drivers/nvme/target/passthru.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/nvme/target/passthru.c b/drivers/nvme/target/passthru.c
index 5f06f4290c52..573a922cd393 100644
--- a/drivers/nvme/target/passthru.c
+++ b/drivers/nvme/target/passthru.c
@@ -482,6 +482,11 @@ u16 nvmet_parse_passthru_admin_cmd(struct nvmet_req *req)
 	}
 }
 
+static void nvme_pt_put_ctrl(struct nvme_ctrl *ctrl)
+{
+	nvme_put_ctrl(ctrl);
+}
+
 int nvmet_passthru_ctrl_enable(struct nvmet_subsys *subsys)
 {
 	struct nvme_ctrl *ctrl;
@@ -532,7 +537,7 @@ int nvmet_passthru_ctrl_enable(struct nvmet_subsys *subsys)
 	return 0;
 
 out_put_ctrl:
-	nvme_put_ctrl(ctrl);
+	nvme_pt_put_ctrl(ctrl);
 out_unlock:
 	mutex_unlock(&subsys->lock);
 	return ret;
@@ -542,7 +547,7 @@ static void __nvmet_passthru_ctrl_disable(struct nvmet_subsys *subsys)
 {
 	if (subsys->passthru_ctrl) {
 		xa_erase(&passthru_subsystems, subsys->passthru_ctrl->cntlid);
-		nvme_put_ctrl(subsys->passthru_ctrl);
+		nvme_pt_put_ctrl(subsys->passthru_ctrl);
 	}
 	subsys->passthru_ctrl = NULL;
 	subsys->ver = NVMET_DEFAULT_VS;
-- 
2.22.1

